includes/gpio.h

```diff
@@ -0,0 +1,84 @@
+#ifndef GPIO_H
+#define GPIO_H
+
+#include <stdint.h>
+
+typedef struct 	GPIO_config_s
+{
+	uint32_t 	GPIO_MODER;  	/* GPIO mode register, 						Address offset 0x00 */
```

Comment:
important: Fields have to be `volatile`.

Also, `GPIO_` prefix is not needed—they are a part of GPIO_config structure anyway.